features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/package-lock.json

```diff
@@ -3087,9 +3087,9 @@
           "integrity": "sha1-hqdj9Y7k18L2sQLkdkBQ3n7ZDGs=",
           "dev": true,
           "requires": {
-            "regenerate": "^1.2.1",
-            "regjsgen": "^0.2.0",
-            "regjsparser": "^0.1.4"
+            "regenerate": "1.4.0",
```

Comment:
IMO committing package-lock.json is wrong if there are no changes going to package.json file. For example say we have ^1.2.1 for given module in package.json. So it can take any 1.x.x version. But if a certain functionality is written to work with 1.2.1 version and if 1.4.0 get locked in the packge-lock.json after a npm install, it might break the functionality.
Open for discussion :)

Reply:
According to what I have read online it is ok to commit the package-lock.json. We should actually lock the module versions in package.json file (by removing ^). I think this would be a better way to prevent from breaking functionality from npm install

Reply:
@dushansilva ,

I think we should not remove the carets from the package.json file, which will make it implicitly a lock file. As I found out, NPM has released a new command to do the installation and update the version locks [npm ci](https://docs.npmjs.com/cli/ci.html) [This blog post](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable).
After searching about this issue , I too agree with the following comment :smile: 

> What's the fucking use of a lock file if you are just going to update it anyways because a package.json dependency can be updated to a newer version?
> 
> And please don't tell me I am suppose to remove the carets from my package.json otherwise I might as well consider that to be our new lock file.

https://github.com/npm/npm/issues/18103#issuecomment-370401935

source: [SO](https://stackoverflow.com/questions/45022048/why-does-npm-install-rewrite-package-lock-json)

Until we figure this out shall we not to commit the package-lock.json file ?
I too agree with @taniamahanama  comment that, updating package versions unknowingly could lead to wired issues.

@taniamahanama , Thanks for bringing this up.
Shall we create a separate git  issue for this in product-apim and move this discussion there, and close this PR ?

Reply:
+1 I will remove the package-lock.json file

Reply:
> According to what I have read online it is ok to commit the package-lock.json. We should actually lock the module versions in package.json file (by removing ^). I think this would be a better way to prevent from breaking functionality from npm install
> 
> https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

@dushansilva : There are scenarios that we should commit package-lock.json. For example if you do a version bump to a particular module, then the package-lock.json will be automatically locked with the newer version. So my concern was about committing it when there are no changes to the package.json file.